_overviews/core/collections-migration-213.md

```diff
@@ -0,0 +1,330 @@
+| `IterableOnce#toIterator` is replaced with `IterableOnce#iterator` | `xs.toIterator` | `xs.iterator` | `NewCollections`, `CrossCompat` |
+| `copyToBuffer` has been deprecated | `xs.copyToBuffer(buffer)` | `buffer ++= xs` | `NewCollections`, `CrossCompat` |
+| `TupleNZipped` has been replaced with `LazyZipN` | `(xs, ys).zipped` | `xs.lazyZip(ys)` | `NewCollections` |
+| `retain` has been renamed to `filterInPlace` | `xs.retain(f)` | `xs.filterInPlace(f.tupled)` | `NewCollections` |
+| `:/` and `/:` operators have been deprecated | `(xs :\ y)(f)` | `xs.foldRight(y)(f)` | `NewCollections`, `CrossCompat` |
+| `companion` operation has been renamed to `iterableFactory` | `xs.companion` | `xs.iterableFactory` |  |
+ 
+## Deprecated things in 2.12 that have been removed in 2.13
+
+- `collection.convert.JavaConversions`. Use `collection.convert.JavaConverters` instead ;
+- `collection.mutable.MutableList` (was not deprecated in 2.12 but was considered to be an implementation detail for implementing other collections). Use an `ArrayDeque` instead, or a `List` and a `var` ;
+- `collection.immutable.Stack`. Use a `List` instead ;
+- `StackProxy`, `MapProxy`, `SetProxy`, `SeqProxy`, etc. No replacement ;
+- `SynchronizedMap`, `SynchronizedBuffer`, etc. Use `java.util.concurrent` instead ;
+
+## Are there new collection types?
+
+`scala.collection.immutable.ArraySeq` is an immutable sequence backed by an array. It is used to pass varargs parameters.
+
+The [`scala-collection-contrib`](https://github.com/scala/scala-collection-contrib) module provides decorators enriching the collections with new operations. You can
+think of this artifact as an incubator: if we get evidence that these operations should be part of the core,
+we might eventually move them.
+
+The following collections are provided:
+
+- `MultiSet` (both mutable and immutable)
+- `SortedMultiSet` (both mutable and immutable)
+- `MultiDict` (both mutable and immutable)
+- `SortedMultiDict` (both mutable and immutable)
+
+## Are there new operations on collections?
+
+The following new partitioning operations are available:
+
+~~~ scala
+def groupMap[K, B](key: A => K)(f: A => B): Map[K, CC[B]] // (Where `CC` can be `List`, for instance)
+def groupMapReduce[K, B](key: A => K)(f: A => B)(g: (B, B) => B): Map[K, B]
+~~~
+
+`groupMap` is equivalent to `groupBy(key).mapValues(_.map(f))`.
+
+`groupMapReduce` is equivalent to `groupBy(key).mapValues(_.map(f).reduce(g))`.
+
+Mutable collections now have transformation operations that modify the collection in place:
+
+~~~ scala
+def mapInPlace(f: A => A): this.type
+def flatMapInPlace(f: A => IterableOnce[A]): this.type
+def filterInPlace(p: A => Boolean): this.type
+def patchInPlace(from: Int, patch: scala.collection.Seq[A], replaced: Int): this.type
+~~~
+
+Another new operation is `distinctBy`:
+
+~~~ scala
+def distinctBy[B](f: A => B): C // (Where `C` can be `List[Int]`, for instance)
+~~~
+
```

Comment:
Another new operation not listed here is `partitionMap`

Reply:
👍